script.js

```diff
@@ -1,406 +1,437 @@
+      // Check if the next item should switch place with the current item:
+      if (b[i].innerHTML.toLowerCase() > b[i + 1].innerHTML.toLowerCase()) {
+        // If next item is alphabetically lower than current item,mark as a switch and break the loop:
+        shouldSwitch = true;
+        break;
+      }
+    }
+    if (shouldSwitch) {
+      //If a switch has been marked, make the switch and mark the switch as done:
+      b[i].parentNode.insertBefore(b[i + 1], b[i]);
+      switching = true;
+    }
+  }
+}
+
+// Sort cooking time. Not working right.
+function sortListTime() {
+  let list, i, switching, b, shouldSwitch;
+  list = document.getElementById("container");
+  switching = true;
+  //Make a loop that will continue until no switching has been done:
+  while (switching) {
+    // Start by saying: no switching is done:
+    switching = false;
+    b = list.getElementsByClassName("time");
+    c = list.getElementsByClassName("card");
+    // Loop through all list items:
+    for (i = 0; i < b.length; i++) {
+      // Start by saying there should be no switching:
+      shouldSwitch = false;
+
+      // Check and replace if a null-value is found.
+      if (b[i].innerText === "null") {
+        b[i].innerText = 0;
+      }
+      // Check if the next item should switch place with the current item:
+      if (parseInt(b[i].innerHTML) > parseInt(b[i + 1].innerHTML)) {
+        // If next item is alphabetically lower than current item, mark as a switch and break the loop:
+        shouldSwitch = true;
+        break;
+      }
+    }
+    if (shouldSwitch) {
+      // If a switch has been marked, make the switch and mark the switch as done:
+      c[i].parentNode.insertBefore(c[i + 1], c[i]);
+      switching = true;
+    }
+  }
+}
+
+// Function to get a random recipe and display it
+const randomRecipe = () => {
+  let randomNumber = Math.floor(Math.random() * recipes.length);
+  console.log(randomNumber);
+  return randomNumber;
+};
+
+// Search bar
+const searchForm = document.getElementById("search-form");
+searchForm.addEventListener("submit", function (event) {
+  // Prevent the form from submitting in the traditional way
+  event.preventDefault();
+  let i;
+  const searchInput = document.getElementById("search-input");
+  // Search user's input to all lower case
+  const searchValue = searchInput.value.toLowerCase();
+  // Search user's input from all recipes in array and Find a recipe that includes user's input text
+  // Check if the recipe name (converted to lowercase) contains the search text
+  const matchedRecipes = recipes.filter((recipe) =>
+    recipe.name.toLowerCase().includes(searchValue)
+  );
+  // Clear existing recipes from the container before loading new ones
+  container.innerHTML = "";
+
+  // Show those recipes only
+  if (matchedRecipes.length > 0) {
+    matchedRecipes.forEach(addRecipeInformation);
+    // Checking for card-divs
+    const y = document.getElementsByClassName("card");
```

Comment:
You already declared that variable in line 255 and could use it from there.

//Wen&Linda